src/librustc_codegen_llvm/common.rs

```diff
@@ -329,10 +329,10 @@ impl ConstMethods<'tcx> for CodegenCx<'ll, 'tcx> {
     fn from_const_alloc(
         &self,
         layout: TyLayout<'tcx>,
-        align: Align,
         alloc: &Allocation,
         offset: Size,
     ) -> PlaceRef<'tcx, &'ll Value> {
+        let align = alloc.align; // follow what CTFE did, not what the layout says
```

Comment:
@oli-obk @eddyb I have no idea if this should rather be `alloc.align` or `layout.align.abi`.

Reply:
I think it's irrelevant right now. The two must be equal, so you can assert that here (the old code for statics bugged out in case the alignments differed, I don't think other code violates this invariant). In the future we may want to dedup certain allocations even if the types used to represent them differ.

Reply:
All right, I added an assertion.